File: VModer.Core/Dto/TraitDto.cs
using System.Text.Json.Serialization;
using VModer.Core.Models.Character;
using VModer.Core.Services;

namespace VModer.Core.Dto;

public sealed class TraitDto
{
    public required string Name { get; init; }
    public required string LocalizedName { get; init; }
    public required string Modifiers { get; init; }
    public required FileOrigin FileOrigin { get; init; }
    public required TraitType Type { get; init; }

    /// <summary>
    /// 特质描述信息
    /// </summary>
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Description { get; set; }
}

File: VModer.Extensions/client/src/dto/TraitDto.ts
import type { DocumentRange } from '../../src/types/DocumentRange';
import type { FileOrigin } from './FileOrigin';
import type { TraitKind } from './TraitKind';
import type { TraitType } from './TraitType';

export interface TraitDto {
  FilePath: string;
  Name: string;
  LocalizedName: string;
  Modifiers: string;
  FileOrigin: FileOrigin;
  GeneralType: TraitType;
  Description?: string;
  Position: DocumentRange;
  /** file:// 形式的 Uri */ 
  IconPath?: string;
  Type: TraitKind
}

export * from './TraitDto';
export * from './FileOrigin';
export * from './TraitType';
export * from './TraitKind';